nemo_aligner/models/mm/stable_diffusion/image_text_rms.py

```diff
@@ -0,0 +1,151 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+import torch
+import torch.nn.functional as F
+from PIL import Image
+from torchvision.transforms import CenterCrop, Compose, Normalize, Resize
+from nemo.collections.multimodal.data.clip.clip_dataset import get_preprocess_fns
+from nemo.collections.multimodal.models.vision_language_foundation.clip.megatron_clip_models import (
+    CLIPTextTransformer,
+    CLIPVisionTransformer,
+    MegatronCLIPModel,
+)
+from nemo.collections.multimodal.parts.utils import setup_trainer_and_model_for_inference
+from nemo.collections.nlp.modules.common.megatron.module import MegatronModule
+
+
+try:
+    from torchvision.transforms import InterpolationMode
+
+    BICUBIC = InterpolationMode.BICUBIC
+except ImportError:
+    BICUBIC = Image.BICUBIC
+
+torch.backends.cuda.matmul.allow_tf32 = True
+
+OPENAI_DATASET_MEAN = (0.48145466, 0.4578275, 0.40821073)
+OPENAI_DATASET_STD = (0.26862954, 0.26130258, 0.27577711)
+
+
+class PickscoreRewardModel(MegatronModule):
+    """CLIP-Based Model"""
+
+    def __init__(self, model_cfg, model_parallel_config, padded_vocab_size, pre_process=True, post_process=True):
+        super(PickscoreRewardModel, self).__init__()
+
+        self.config = model_parallel_config
+        self.pre_process = pre_process
+        self.post_process = post_process
+        self.vision_encoder = CLIPVisionTransformer(
+            model_cfg.vision, model_parallel_config, pre_process=self.pre_process, post_process=self.post_process,
+        )
+        self.text_encoder = CLIPTextTransformer(
+            model_cfg.text,
+            model_parallel_config,
+            padded_vocab_size,
+            pre_process=self.pre_process,
+            post_process=self.post_process,
+        )
+
+        self.logit_scale = torch.nn.Parameter(torch.ones([]) * np.log(1 / 0.07))
+
+    def set_input_tensor(self, input_tensor):
+        """See megatron.model.transformer.set_input_tensor()"""
+        # TODO (yuya): fix this
+        pass
+
+    def get_reward(self, images, captions):
+
+        text_features = self.text_encoder(captions)
+        image_features = self.vision_encoder(images)
+
+        rewards = (
+            self.logit_scale.exp()
+            * torch.matmul(F.normalize(image_features, dim=-1), F.normalize(text_features, dim=-1).t()).diag()
+        )
+
+        return rewards
+
+
+class MegatronCLIPRewardModel(MegatronCLIPModel):
+    def __init__(self, cfg, trainer):
+        super().__init__(cfg, trainer)
+        self.differentiable_preprocess = self.diff_preprocess()
+
+    def diff_preprocess(self):
+
+        return Compose(
+            [
+                Resize(224, interpolation=BICUBIC, antialias=True),
+                CenterCrop(224),
+                self.rescale,
+                Normalize(OPENAI_DATASET_MEAN, OPENAI_DATASET_STD),
```

Comment:
can we push these 2 constants into the model config?

Reply:
added to init as we discussed